st2actions/st2actions/scheduler.py

```diff
@@ -0,0 +1,75 @@
+# Licensed to the StackStorm, Inc ('StackStorm') under one or more
+# contributor license agreements.  See the NOTICE file distributed with
+# this work for additional information regarding copyright ownership.
+# The ASF licenses this file to You under the Apache License, Version 2.0
+# (the "License"); you may not use this file except in compliance with
+# the License.  You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from kombu import Connection
+from oslo.config import cfg
+
+from st2common import log as logging
+from st2common.constants import action as action_constants
+from st2common.exceptions.db import StackStormDBObjectNotFoundError
+from st2common.services import executions
+from st2common.persistence.action import LiveAction
+from st2common.transport import consumers, liveaction
+from st2common.util import action_db as action_utils
+from st2common.util import system_info
+
+
+LOG = logging.getLogger(__name__)
+
+ACTIONRUNNER_REQUEST_Q = liveaction.get_status_management_queue(
+    'st2.actionrunner.req', routing_key=action_constants.LIVEACTION_STATUS_REQUESTED)
+
+
+class ActionExecutionScheduler(consumers.MessageHandler):
+    def process(self, request):
+        if request.status != action_constants.LIVEACTION_STATUS_REQUESTED:
```

Comment:
One thing we want to do (in another PR) is to have these classes define the `type` of request object and do validation.